### PR TITLE
[Bug] Recruiting Pipeline applicant schedule meeting doesn't update to show the 'next' meeting

### LINKED ISSUE #633: [Bug] Recruiting Pipeline applicant schedule meeting doesn't update to show the 'next' meeting
once you set the org date for a meeting then go and schedule again (not sure if its suppose to edit the org one or create a new one completely)  with a date sooner than the org date entered, it doesn't update

### PR DESCRIPTION
**Pull request by JavaLavaMT**
closes #633

### CODE REVIEW

File: app/components/card-pipeline/stage-card.js [hunk 1]
```diff
@@ -21,8 +21,10 @@ const PipelineCardComponent = Component.extend({
     return (this.get('stages') || []).filter(stage => get(stage, '_id') !== stageId);
   }),
 
-  nextMeeting: computed('application.[]', function () {
-    return this.get('getNextMeeting').perform();
+  nextMeeting: computed('application.[]', 'newScheduledMeeting', function () {
+    if (!this.meetingFetched || this.application.id !== this.newScheduleMeeting) {
```
Comment (James1x0) [inline app/components/card-pipeline/stage-card.js]: Are you sure this is working?
```suggestion
    if (!this.meetingFetched || this.application.id !== this.newScheduledMeeting) {
```

File: app/components/card-pipeline/stage-card.js [hunk 2]
```diff
@@ -33,12 +35,11 @@ const PipelineCardComponent = Component.extend({
     try {
       let results = yield this.get('store').query('event', {
         contextType: 'JobApplication',
-        contextId:   this.get('application.id'),
-        limit:       1,
-        start:       { $gt: new Date() },
-        sort:        { start: -1 }
+        contextId:   this.get('application.id')
```
Comment (James1x0) [inline app/components/card-pipeline/stage-card.js]: Why delete?

Comment (JavaLavaMT) [reply app/components/card-pipeline/stage-card.js]: if im right in think so, there is no more need as we are only making one event so we will only have one file come back based on this query with the contextId. or did i miss something

File: app/components/card-pipeline/stage-card.js (hunk 1 above)
Comment (JavaLavaMT) [reply app/components/card-pipeline/stage-card.js]: yeah it works just fine, it just calls the query everytime no matter what :) NAI

Comment (JavaLavaMT) [reply app/components/card-pipeline/stage-card.js]: fixed

File: app/components/card-pipeline/stage-card.js (hunk 2 above)
Comment (James1x0) [reply app/components/card-pipeline/stage-card.js]: @JavaLavaMT What if the meeting passes and we schedule a second interview...

File: app/controllers/account/job-opening/campaign/applicant-tracking.js [hunk 3]
```diff
@@ -38,7 +38,7 @@ export default Controller.extend(addEdit, ajaxStatus, modalSupport, {
 
   resetMeeting () {
     if (this.get('currentMeeting')) {
-      this.get('currentMeeting').destroy();
+      this.set('currentMeeting', undefined);
```
Comment (James1x0) [inline app/controllers/account/job-opening/campaign/applicant-tracking.js]: We need both of these to happen. We need to remove the record from ember data's store and set the controller's property.

File: app/controllers/account/job-opening/campaign/applicant-tracking.js [hunk 4]
```diff
@@ -76,26 +76,41 @@ export default Controller.extend(addEdit, ajaxStatus, modalSupport, {
...
-  saveMeeting (event) {
+  async saveMeeting (event) {
     this.ajaxStart();
 
     const app = this.get('appInScheduler'),
           isEmployeeApplicant = app.get('isEmployee');
 
+    let previousMeeting = await this.store.queryRecord('event', {
+      contextType: 'JobApplication',
+      attendantId: get(isEmployeeApplicant ? app.get('employee') : app.get('applicant'), 'id')
```
Comment (James1x0) [inline app/controllers/account/job-opening/campaign/applicant-tracking.js]:
```suggestion
      attendantId: get(app.get(isEmployeeApplicant ? 'employee' : 'applicant'), 'id')
```

File: app/controllers/account/job-opening/campaign/applicant-tracking.js [hunk 5]
```diff
@@ -76,26 +76,41 @@ export default Controller.extend(addEdit, ajaxStatus, modalSupport, {
...
-    event.save()
-    .then(meeting => {
       const title = meeting.get('title') ? `"${meeting.get('title')}"` : 'meeting',
             start = moment(meeting.get('start'));
 
       this.ajaxSuccess(`Scheduled ${title} at ${start.format('h:mma [on] M/D/YY')}`);
-    });
+
+      this.set('newScheduledMeeting', this.get('newScheduledMeeting') !== app.get('id') ? app.get('id') : false);
+
```
Comment (James1x0) [inline app/controllers/account/job-opening/campaign/applicant-tracking.js]:
```suggestion
```

File: app/components/card-pipeline/stage-card.js [hunk 6]
```diff
@@ -32,13 +34,15 @@ const PipelineCardComponent = Component.extend({
   getNextMeeting: task(function*() {
     try {
       let results = yield this.get('store').query('event', {
-        contextType: 'JobApplication',
-        contextId:   this.get('application.id'),
         limit:       1,
         start:       { $gt: new Date() },
-        sort:        { start: -1 }
+        sort:        { start: 1 },
```
Comment (James1x0) [inline app/components/card-pipeline/stage-card.js]: Don't we want the soonest first?

File: app/controllers/account/job-opening/campaign/applicant-tracking.js [hunk 7]
```diff
@@ -88,14 +88,18 @@ export default Controller.extend(addEdit, ajaxStatus, modalSupport, {
...
 
-    event.save()
-    .then(meeting => {
       const title = meeting.get('title') ? `"${meeting.get('title')}"` : 'meeting',
             start = moment(meeting.get('start'));
 
       this.ajaxSuccess(`Scheduled ${title} at ${start.format('h:mma [on] M/D/YY')}`);
-    });
+
+      this.set('newScheduledMeeting', app.get('id'));
```
Comment (James1x0) [inline app/controllers/account/job-opening/campaign/applicant-tracking.js]: To fix the bug where you wouldn't refresh if scheduling a sooner meeting for a job application, you can set `newScheduledMeeting` to `null` before you make the request to save.

**Pull request merged**